src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -416,6 +338,36 @@ public virtual Expression BindCollectionResourceCastNode(CollectionResourceCastN
             return OfType(source, clrType);
         }
 
+        /// <summary>
+        /// Binds a <see cref="CollectionConstantNode"/> to create a LINQ <see cref="Expression"/> that
+        /// represents the semantics of the <see cref="CollectionConstantNode"/>.
+        /// </summary>
+        /// <param name="node">The node to bind.</param>
+        /// <returns>The LINQ <see cref="Expression"/> created.</returns>
+        public virtual Expression BindCollectionConstantNode(CollectionConstantNode node)
+        {
+            // It's fine if the collection is empty; the returned value will be an empty list.
+            ConstantNode firstNode = node.Collection.FirstOrDefault();
+            object value = null;
+            if (firstNode != null)
+            {
+                value = firstNode.Value;
+            }
+
+            Type constantType = RetrieveClrTypeForConstant(node.ItemType, ref value);
+            Type listType = typeof(List<>).MakeGenericType(constantType);
+            IList castedList = Activator.CreateInstance(listType) as IList;
```

Comment:
Create a object directly? 

or just use "Expression.New(…)"

Reply:
Thanks for the suggestion. I tried various ways to create the object directly but those methods seem to require knowledge of the type at runtime. I've reviewed ways to use Expression.New and NewArrayExpression but due to their setup, those methods seem to have more complexity than the code here. I will leave this as-is.